source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -30,9 +32,10 @@ class SimpleLookupContext : public LookupContext {
     cb(std::make_unique<Buffer::OwnedImpl>(&body_[range.begin()], range.length()));
   }
 
-  void getTrailers(LookupTrailersCallback&&) override {
-    // TODO(toddmgreer): Support trailers.
-    NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
+  void getTrailers(LookupTrailersCallback&& cb) override {
+    if (trailers_ != nullptr) {
+      cb(Http::createHeaderMap<Http::ResponseTrailerMapImpl>(*trailers_));
+    }
```

Comment:
it is unexpected that cb() would simply not be called if trailers is null. In contrast, for the getHeaders case, cb is called regardless, though sometimes wiht an empty LookupResult (that also seems potentially confusing).

Maybe it'd be more obvious what the semantics are to call with a std::optional<LookupResult> in both cases?

Reply:
I implemented the same check as in getHeaders for consistency, but I could change the semantics in the subsequent CL if you think it would still be worth it.